### PR TITLE
Refactor consolidate import from io in providers

### PR DESCRIPTION
**Pull request by eumiro**

### CODE REVIEW

File: airflow/providers/docker/operators/docker.py [hunk 1]
```diff
@@ -451,11 +451,11 @@ def copy_from_docker(container_id, src):
...
-            file_standin = BytesIO(b"".join(archived_result))
-            tar = tarfile.open(fileobj=file_standin)
-            file = tar.extractfile(stat["name"])
-            lib = getattr(self, "pickling_library", pickle)
-            return lib.loads(file.read())
+            with BytesIO(b"".join(archived_result)) as f:
+                tar = tarfile.open(fileobj=f)
+                file = tar.extractfile(stat["name"])
+                lib = getattr(self, "pickling_library", pickle)
+                return lib.loads(file.read())
```
Comment (uranusjr) [inline airflow/providers/docker/operators/docker.py]:
```suggestion
                return lib.load(file)
```

I wonder if this is possible

File: airflow/providers/google/cloud/hooks/gcs.py [hunk 2]
```diff
@@ -566,10 +566,10 @@ def _call_with_retry(f: Callable[[], None]) -> None:
             if gzip:
                 if isinstance(data, str):
                     data = bytes(data, encoding)
-                out = BytesIO()
-                with gz.GzipFile(fileobj=out, mode="w") as f:
-                    f.write(data)
-                data = out.getvalue()
+                with BytesIO() as out:
+                    with gz.GzipFile(fileobj=out, mode="w") as f:
+                        f.write(data)
```
Comment (uranusjr) [inline airflow/providers/google/cloud/hooks/gcs.py:571]:
```suggestion
                with BytesIO() as out, gz.GzipFile(fileobj=out, mode="w") as f:
                    f.write(data)
```

File: airflow/providers/google/cloud/hooks/gcs.py [hunk 3]
```diff
@@ -566,10 +566,10 @@ def _call_with_retry(f: Callable[[], None]) -> None:
...
+                    data = out.getvalue()
```
Comment (uranusjr) [inline airflow/providers/google/cloud/hooks/gcs.py]:
```suggestion
                with BytesIO() as out, gz.GzipFile(fileobj=out, mode="w") as f:
                    f.write(data)
                    data = out.getvalue()
```

I believe this works.

**Comment by uranusjr** [comment]
Not sure if we want to use `from` or `import io` here.

File: airflow/providers/docker/operators/docker.py (hunk 1 above)
Comment (eumiro) [reply airflow/providers/docker/operators/docker.py]: Let me try it.

**Review by vincbeck — APPROVED** [review]

**Review by o-nikolas — APPROVED** [review]

**Comment by vincbeck** [comment]
@uranusjr ?

**Pull request merged**